src/pyj/read_book/viewport.pyj

```diff
@@ -52,6 +70,29 @@ class ScrollViewport:
     def height(self):
         return window.innerHeight
 
+    # Assure that the viewport position returned is corrected for the RTL
+    # mode of ScrollViewport.
+    def viewport_to_document(self, x, y, doc):
+        self.__reset_transforms()
+
+        # Convert x, y from the viewport (window) co-ordinate system to the
+        # document (body) co-ordinate system
+        doc = doc or window.document
+        topdoc = window.document
+        while doc is not topdoc:
+            # We are in a frame
+            frame = doc.defaultView.frameElement
+            rect = frame.getBoundingClientRect()
+            x += rect.left
+            y += rect.top
+            doc = frame.ownerDocument
+        win = doc.defaultView
+        wx, wy = win.pageXOffset, win.pageYOffset
+        x += wx
+        y += wy
+        if self.rtl:
+            return -x, y
```

Comment:
This would be the place the transform happens. I removed two copies of viewport_to_document from the codebase, so I consider the reduction of duplicate code a win. I hope you agree!